### PR TITLE
Removed createTouch and createTouchList

### LINKED ISSUE #25978: Remove createTouch and createTouchList
These were removed from the spec in https://github.com/w3c/touch-events/pull/96. We should delete https://github.com/servo/servo/blob/f5c086b30759e7203bb1ab412693f0e86252dafc/components/script/dom/webidls/Document.webidl#L189-L200 and https://github.com/servo/servo/blob/f5c086b30759e7203bb1ab412693f0e86252dafc/components/script/dom/document.rs#L3897-L3918.

### PR DESCRIPTION
**Pull request by akshatagarwl**
---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #25978

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because ___

### CODE REVIEW

**Comment by jdm** [comment]
We correctly pass the test that verifies that these APIs don't exist now:
```
  ▶ Unexpected subtest result in /touch-events/historical.html:
  └ PASS [expected FAIL] Document::createTouch
  ▶ Unexpected subtest result in /touch-events/historical.html:
  └ PASS [expected FAIL] Document::createTouchList
```
Please [update the test results](https://github.com/servo/servo/blob/master/tests/wpt/README.md#updating-test-expectations) for these tests in tests/wpt/metadata/touch-events/historical.html.ini!

File: tests/wpt/metadata/touch-events/historical.html.ini [hunk 1]
```diff
@@ -12,11 +12,11 @@
     expected: FAIL
 
   [Document::createTouch]
-    expected: FAIL
```
Comment (jdm) [inline tests/wpt/metadata/touch-events/historical.html.ini]: Both of these lines can be removed instead.

File: tests/wpt/metadata/touch-events/historical.html.ini [hunk 2]
```diff
@@ -12,11 +12,11 @@
...
+    expected: PASS
 
   [TouchList::identifiedTouch]
     expected: FAIL
 
   [Document::createTouchList]
-    expected: FAIL
+    expected: PASS
```
Comment (jdm) [inline tests/wpt/metadata/touch-events/historical.html.ini]: Same here.

**Review by jdm — CHANGES_REQUESTED** [review]

**Comment by humancalico** [comment]
@jdm Can you have a look at it now?

**Comment by jdm** [comment]
Looks good! Please squash them into a single commit and we'll merge them!

**Comment by jdm** [comment]
@bors-servo r+
Thanks!

**Comment by jdm** [comment]
@bors-servo retry
#23290

**Comment by jdm** [comment]
@bors-servo retry
#26020 #23290

**Comment by CYBAI** [comment]
@bors-servo retry

- #25513

**Pull request merged**